frinx/common/worker/worker.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import TypeAlias
+
+from pydantic import ValidationError
+from pydantic.dataclasses import dataclass
+
+from frinx.client.frinx_conductor_wrapper import FrinxConductorWrapper
+from frinx.common.conductor_enums import TaskResultStatus
+from frinx.common.util import jsonify_description
+from frinx.common.util import snake_to_camel_case
+from frinx.common.worker.task import Task
+from frinx.common.worker.task_def import BaseTaskdef
+from frinx.common.worker.task_def import DefaultTaskDefinition
+from frinx.common.worker.task_def import TaskDefinition
+from frinx.common.worker.task_def import TaskInput
+from frinx.common.worker.task_def import TaskOutput
+from frinx.common.worker.task_result import TaskResult
+
+logger = logging.getLogger(__name__)
+
+RawTaskIO: TypeAlias = dict[str, Any]
+TaskExecLog: TypeAlias = str
+
+
+class Config:
+    arbitrary_types_allowed = True
+    alias_generator = snake_to_camel_case
+    allow_population_by_field_name = True
+
+
+@dataclass(config=Config)
+class WorkerImpl(ABC):
+    task_def: TaskDefinition = None # type: ignore
+    task_def_template: type[BaseTaskdef] | type[DefaultTaskDefinition] = None # type: ignore
```

Comment:
None default value for a variable that should not allow None. We should fix the type issues in this class.